app/presenters/keyword_presenter.rb

```diff
@@ -12,6 +12,18 @@ def matching_adword_urls
     return unless defined?(filter_params[:adwords_url_contains]) && filter_params[:adwords_url_contains]
```

Comment:
I am not sure why do we need to check `defined?` here? If you would like to check `nil` you can use `nil?` instead. 

https://ruby-doc.org/docs/keywords/1.9/Object.html#method-i-defined-3F

Reply:
I got NoMethodError when trying to I used `present?` or `any?` when the `filter_params` is nil. So trying to access the  param `adwords_url_contains` in a nil object will throw an error.

Reply:
You can avoid this with a default value: `def initialize(keyword, filter_params = {})`.

```suggestion
    return unless filter_params[:adwords_url_contains]
```